tests/aws/scenario/kinesis_firehose/conftest.py

```diff
@@ -1,47 +1,45 @@
 import json
+import logging
 
 from localstack.utils.sync import retry
 
+LOG = logging.getLogger(__name__)
 
-def read_s3_data(aws_client, bucket_name: str, timeout: int = 10) -> dict[str, str]:
-    def _get_data():
-        response = aws_client.s3.list_objects(Bucket=bucket_name)
-        if response.get("Contents") is None:
-            raise Exception("No data in bucket yet")
 
-        keys = [obj.get("Key") for obj in response.get("Contents")]
+def read_s3_data(aws_client, bucket_name: str) -> dict[str, str]:
+    response = aws_client.s3.list_objects(Bucket=bucket_name)
+    if response.get("Contents") is None:
+        raise Exception("No data in bucket yet")
 
-        bucket_data = dict()
-        for key in keys:
-            response = aws_client.s3.get_object(Bucket=bucket_name, Key=key)
-            data = response["Body"].read().decode("utf-8")
-            bucket_data[key] = data
-        return bucket_data
+    keys = [obj.get("Key") for obj in response.get("Contents")]
 
-    bucket_data = retry(_get_data, sleep=1, retries=timeout)
+    bucket_data = dict()
+    for key in keys:
+        response = aws_client.s3.get_object(Bucket=bucket_name, Key=key)
+        data = response["Body"].read().decode("utf-8")
+        bucket_data[key] = data
     return bucket_data
 
 
-def get_all_messages_from_s3(
+def get_all_expected_messages_from_s3(
     aws_client,
     bucket_name: str,
-    timeout: int = 300,
     sleep: int = 5,
     retries: int = 3,
-    expected_message_count: int = 4,
+    expected_message_count: int | None = None,
 ) -> list[str]:
-    # poll file from s3 buckets
     def get_all_messages():
-        s3_data_bucket = read_s3_data(aws_client, bucket_name, timeout=timeout)
+        bucket_data = read_s3_data(aws_client, bucket_name)
         messages = []
-        for input_string in s3_data_bucket.values():
+        for input_string in bucket_data.values():
             json_array_string = "[" + input_string.replace("}{", "},{") + "]"
             message = json.loads(json_array_string)
+            LOG.debug(f"Received messages: {message}")
             messages.extend(message)
-        if len(messages) != expected_message_count:
+        if not expected_message_count and len(messages) != expected_message_count:
```

Comment:
```suggestion
        if expected_message_count is not None and len(messages) != expected_message_count:
```

?

Reply:
fixed in: [216d0b8](https://github.com/localstack/localstack/pull/10243/commits/216d0b8bad510fe8e515e89a6233facc3fbe0243)